include/oneapi/dpl/internal/binary_function_impl.h

```diff
@@ -0,0 +1,45 @@
+/*
+ *  Copyright (c) Intel Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#ifndef _ONEDPL_BINARY_FUNCTION_IMPL
+#define _ONEDPL_BINARY_FUNCTION_IMPL
+
+namespace oneapi
+{
+namespace dpl
+{
+
+#if __cplusplus > 201402L
+template <class _Arg1, class _Arg2, class _Result>
```

Comment:
To be honest, I don't understand the reason why we introduce this functionality. It was deprecated in C++11 and removed in C++17. Without a good reason I think we should not support such a legacy, especially it's purely compile-time structure.